### ISSUE TITLE
dev: refactor estimate_fee in pallets/starknet/src/lib.rs

### ISSUE THREAD

**Issue by lana-shanghai** [keep-starknet-strange/madara#1315]
## Description 

Currently in `estimate_fee` we [iterate](https://github.com/keep-starknet-strange/madara/blob/main/crates/pallets/starknet/src/lib.rs#L1102) with a `for` loop over transactions, but it is better to use `.into_iter()` and `map_err` and then `collect` the results into a vector, in order to optimize the function and stop the computation if one the results errors. Similar refactoring already [takes place](https://github.com/keep-starknet-strange/madara/blob/main/crates/client/rpc/src/lib.rs#L900) in the RPC: 

```
let transactions =
    request.into_iter().map(|tx| tx.try_into()).collect::<Result<Vec<UserTransaction>, _>>().map_err(|e| {
        error!("Failed to convert BroadcastedTransaction to UserTransaction: {e}");
        StarknetRpcApiError::InternalServerError
    })?;
```

**Comment by Yogalholic** [comment]
Hello, I would like to give it a try please.

**Comment by tdelabro** [comment]
closed by #1307

**Comment by tdelabro** [comment]
This is the original comment: https://github.com/keep-starknet-strange/madara/pull/1240#discussion_r1387930853

**Comment by tdelabro** [comment]
all yours @Yogalholic

**Comment by lana-shanghai** [comment]
Update: 

see loops that need refactoring [here](https://github.com/keep-starknet-strange/madara/blob/main/crates/pallets/starknet/src/lib.rs#L1107) and [here](https://github.com/keep-starknet-strange/madara/blob/main/crates/pallets/starknet/src/lib.rs#L1146)

**Comment by Yogalholic** [comment]
I'm still working on the issue and opened a draft PR. I think we need a `map()` between `into_iter()` and `map_err()`. I've tried using instead a `try_for_each()` or a collect::<Result<Value, `E>>` (as mentioned in [#1240 (comment)](https://github.com/keep-starknet-strange/madara/pull/1240#discussion_r1387930853)) but I've not successfully solved the issue yet.

**Comment by tdelabro** [comment]
fixed by #1382
